server/tests/services/datacommons_test.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import unittest
+from unittest import mock
+
+from server.services.datacommons import v2node_paginated
+
+MODULE_DIR = os.path.dirname(os.path.abspath(__file__))
+TEST_DATA_DIR = os.path.join(MODULE_DIR, "..", "test_data", "datacommons")
+
+
+def get_json(filename):
+  if not filename.lower().endswith(".json"):
+    filename += ".json"
+  filepath = os.path.join(TEST_DATA_DIR, filename)
+  with open(filepath, "r") as f:
+    return json.load(f)
+
+
+class TestServiceDataCommonsV2NodePaginated(unittest.TestCase):
+
+  @mock.patch('server.services.datacommons.post')
+  def test_termination_condition_max_pages_fetched(self, mock_post):
+    response_with_next_token = get_json('v2node_response_with_next_token')
+    mock_post.side_effect = [response_with_next_token, response_with_next_token]
```

Comment:
should also check that the request going to the mock post request is correct by only returning mocked response  for the specific request like: https://github.com/datacommonsorg/website/blob/master/server/tests/routes/api/series_test.py#L214-L245

Reply:
added input checks in side_effect functions. Is it okay to asserts like how I did it?